### PR TITLE
(refactor) O3-3813: Refactor input fields in ResultsFormField component

### PR DESCRIPTION
**Pull request by jwnasambu**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

I fixed the code to ensure TextInput component where concept.datatype.display is "Numeric" to use type="number"
and the TextInput component where concept.datatype.display is "Text" uses type="textInput"

## Screenshots

https://github.com/user-attachments/assets/636add88-bbcb-4236-a344-e27adb3f8d73

## Related Issue
https://openmrs.atlassian.net/browse/O3-3813

### CODE REVIEW

File: src/results/result-form-field.component.tsx [hunk 1]
```diff
@@ -14,22 +14,23 @@ interface ResultFormFieldProps {
...
+  const isCoded = (concept: ConceptReference) => concept.datatype?.display === 'Coded';
+  const isPanel = (concept: ConceptReference) => concept.setMembers?.length > 0;
 
   const printValueRange = (concept: ConceptReference) => {
-    if (concept?.datatype?.display === 'Numeric') {
-      const maxVal = Math.max(concept?.hiAbsolute, concept?.hiCritical, concept?.hiNormal);
-      const minVal = Math.min(concept?.lowAbsolute, concept?.lowCritical, concept?.lowNormal);
-      return ` (${minVal ?? 0} - ${maxVal > 0 ? maxVal : '--'} ${concept?.units ?? ''})`;
+    if (isNumeric(concept)) {
+      const maxVal = Math.max(concept?.hiAbsolute ?? 0, concept?.hiCritical ?? 0, concept?.hiNormal ?? 0);
```
[the remark this answers is not in the document]
Comment (ibacher) [reply src/results/result-form-field.component.tsx:24]: Yeah, so I think the idea in the ticket itself is fine, but I would resist the urge to render any sort of range, unless and only if both a `hiAbsolute` and `lowAbsolute` are set. For most lab values, we do not have these set.

File: src/results/result-form-field.component.tsx [hunk 2]
```diff
@@ -14,22 +14,23 @@ interface ResultFormFieldProps {
...
+      const minVal = Math.min(concept?.lowAbsolute ?? 0, concept?.lowCritical ?? 0, concept?.lowNormal ?? 0);
+      return `(${minVal} - ${maxVal > 0 ? maxVal : '--'} ${concept?.units ?? ''})`;
     }
     return '';
   };
 
   return (
     <>
-      {isTextOrNumeric(concept) && (
+      {isText(concept) && (
```
Comment (ibacher) [inline src/results/result-form-field.component.tsx:33]: Shouldn't we have something here for `isNumeric`?

File: src/results/result-form-field.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibacher) [reply src/results/result-form-field.component.tsx:24]: Btw, using critical or normal values as a range is definitely not the right thing. Those are thresholds, not hard limits.

**Comment by jwnasambu** [comment]
@ojwanganto and @ibacher kindly I have fixed the proposed changes. Kindly feel free to review my PR at your convenience time, please!

File: src/results/result-form-field.component.tsx [hunk 3]
```diff
@@ -29,20 +32,39 @@ const ResultFormField: React.FC<ResultFormFieldProps> = ({ concept, control }) =
...
+            <NumberInput
+              allowEmpty
+              className={styles.numberInput}
+              disableWheel
+              hideSteppers
+              id={concept.uuid}
+              key={concept.uuid}
+              label={concept?.display + printValueRange(concept)}
+              onChange={(event) => field.onChange(event.target.value)}
+              step={0.01}
```
Comment (denniskigen) [inline src/results/result-form-field.component.tsx:66]: @ibacher @ojwanganto is a step precision value of 0.01 for these number inputs reasonable?

Comment (ojwanganto) [reply src/results/result-form-field.component.tsx:66]: This doesn't look appropriate for all use cases. If we can compute this based on the concept in question then we can have much control. For instance, a test that accepts decimals should infer a lot from the configured min and max values but we can hear @ibacher's thoughts

**Comment by denniskigen** [comment]
https://github.com/openmrs/openmrs-esm-laboratory-app/pull/87 effectively made the LabResultsOrder form the source of truth for recording lab results. Therefore, these changes need to be made in the Orders app instead. I've filed a PR doing just that [here](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1985/files).

**Pull request closed without merging**